tools/unicode_properties_parse/format_width_estimate_intervals.py

```diff
@@ -4,57 +4,73 @@
 # The following code generates data for _Width_estimate_intervals_v2 in <format>.
 
 import re
-from dataclasses import dataclass
+from enum import Enum
 from typing import TextIO
 from pathlib import Path
 
 
-class width_u:
-    is_1: bool = False
-    is_2: bool = True
+LINE_REGEX = re.compile(r"([0-9A-Z]+)(\.\.[0-9A-Z]+)?;(A|F|H|N|Na|W) *#.*")
 
 
-class table_u:
-    # A valid Unicode code point won't exceed max_u.
-    max_u: int = 0x10FFFF
+class UnicodeWidth(Enum):
+    IS_1: int = 1
+    IS_2: int = 2
+
+
+class UnicodeTable:
+    # A valid Unicode code point won't exceed MAX_UNICODE_POINT.
+    MAX_UNICODE_POINT: int = 0x10FFFF
+    UNICODE_TABLE_SIZE: int = MAX_UNICODE_POINT + 1
 
     def __init__(self):
-        self.table = [width_u.is_1] * (self.max_u + 1)
+        self.table = [UnicodeWidth.IS_1] * (self.UNICODE_TABLE_SIZE)
 
     def fill_range(self, rng: tuple, width: bool):
-        from_, to_ = rng[0], rng[1]
+        from_, to_ = rng
         assert from_ <= to_, "impl assertion failed"
-        assert to_ <= self.max_u, "impl assertion failed"
+        assert to_ <= self.MAX_UNICODE_POINT, "impl assertion failed"
         self.table[from_ : to_ + 1] = [width] * (to_ - from_ + 1)
 
     def print_intervals(self):
-        # Print table for _Width_estimate_intervals_v2.
-        c = 0
-        last = self.table[0]
-        for u, el in enumerate(self.table):
-            if el != last:
+        """
+        Generates _Width_estimate_intervals_v2.
+        That is, starting from the second code point ([1])
+        until and including the last code point([MAX_UNICODE_POINT]),
+        for code point [U], if width[U]!=width[U-1], we print I to indicate the new range.
+        """

```

Comment:
Emm, I was explaining the intent of these functions, they are not polished enough to serve as comments directly :|

Reply:
>I'll try to think of a more accurate version, please give me some time

Sure. We can change the comments more.